src/exoticatechnologies/modifications/exotics/impl/HullmodExotic.kt

```diff
@@ -206,20 +205,32 @@ open class HullmodExotic(
      */
     private fun unapplyExoticHullmodAndRemoveExoticaAndHullmod(member: FleetMemberAPI, moduleVariant: ShipVariantAPI, optionalMemberMods: Optional<ShipModifications>) {
         // If optional is there
-        if (optionalMemberMods.isPresent()) {
+        val hasAnyMods = if (optionalMemberMods.isPresent()) {
             val memberMods = optionalMemberMods.get()
             memberMods.removeExotic(this@HullmodExotic)
             ShipModLoader.set(member, moduleVariant, memberMods)
+
+            // And return if we have any mods (exotics or upgrades) left
+            memberMods.hasAnyModifications()
         } else {
             // Otherwise, extract them manually and use them
-            val memberMods = get(member, moduleVariant)
+//            val memberMods = ShipModLoader.get(member, moduleVariant)   //TODO fix
+            val memberMods = ShipModLoader.getFromVariant(moduleVariant)
```

Comment:
then consider using `HullmodExoticHandler.getCorrectMods(member, moduleVariant)` here, however i remember that one not working(?).

needs two separate test runs.

Reply:
Nope, as mentioned - it breaks uninstalling AlphaSubcore, the children modules are left with un-restored OP costs and have a lower OP instead of full.

should not change.

Reply:
nope, not touching this. it's flaky enough as it is from the Refit screen already.